test/unix/main.ml

```diff
@@ -286,59 +286,33 @@ module Close = struct
     check_equivalence ro tbl;
     Index.close ro
 
-  let test_read_after_close t k tbl =
-    check_equivalence t tbl;
-    Index.close t;
-    Alcotest.check_raises "Read after close raises Not_found" Not_found
-      (fun () -> ignore (Index.find t k))
-
-  let test_read_after_close_readonly t k tbl =
-    check_equivalence t tbl;
-    Index.close t;
-    let exn = Unix.Unix_error (Unix.EBADF, "read", "") in
-    Alcotest.check_raises "Cannot read in readonly index after close." exn
-      (fun () -> ignore (Index.find t k))
-
-  let fail_read_after_close () =
-    let Context.{ rw; tbl; _ } = Context.full_index () in
-    let k, v = (Key.v (), Value.v ()) in
-    Index.replace rw k v;
-    Hashtbl.replace tbl k v;
-    test_read_after_close rw k tbl
-
-  let fail_write_after_close () =
-    let Context.{ rw; _ } = Context.empty_index () in
-    Index.close rw;
-    let k, v = (Key.v (), Value.v ()) in
-    (* a single add does not fail*)
-    Index.replace rw k v;
-    let exn = Unix.Unix_error (Unix.EBADF, "read", "") in
-    Alcotest.check_raises "Cannot write in index after close." exn (fun () ->
-        Hashtbl.iter (fun k v -> Index.replace rw k v) main.tbl)
-
-  let open_twice () =
-    let Context.{ rw; tbl; clone } = Context.full_index () in
-    let rw2 = clone ~readonly:false in
-    let k, v = (Key.v (), Value.v ()) in
-    Index.replace rw k v;
-    Hashtbl.replace tbl k v;
-    Index.close rw;
-
-    (* while another instance is still open, read does not fail*)
-    check_equivalence rw tbl;
-    test_read_after_close rw2 k tbl
-
-  let open_twice_readonly () =
-    let Context.{ rw; tbl; clone } = Context.full_index () in
-    let k, v = (Key.v (), Value.v ()) in
-    Index.replace rw k v;
-    Hashtbl.replace tbl k v;
-    Index.close rw;
-    let ro1 = clone ~readonly:true in
-    let ro2 = clone ~readonly:true in
-    check_equivalence ro1 tbl;
-    Index.close ro1;
-    test_read_after_close_readonly ro2 k tbl
+  let fail_api_after_close () =
+    let Context.{ clone; _ } = Context.full_index () in
+    let k = Key.v () in
+    let v = Value.v () in
+    let calls t =
+      [
+        ("clear", fun () -> ignore (Index.clear t));
+        ("find", fun () -> ignore (Index.find t k));
```

Comment:
```suggestion
        ("find", fun () -> ignore (Index.find t k : string));
```
These tests will silently weaken if we were to e.g. change to Lwt or add extra parameters.

A lot of these ignores aren't necessary since they ignore a unit. For `find` and `mem`, I think you should assert the types of the ignored values.

Reply:
I think these are already pushed on master [commit](https://github.com/mirage/index/commit/33eca734b570a3fd7036f33196a19d4d8866ac39). 
@pascutto can you do a rebase?

Reply:
Yep; I was being daft. I've made these suggestions in a separate PR now... #112.